### PR TITLE
Updating all Python package requirements.txt files

### PR DESCRIPTION
**Pull request by CatGarab**
The intention is that you can install all dependencies by running `pip3 install -r requirements.txt` from the repo's root directory

Most packages are version locked based on the version being used by Buildroot

### CODE REVIEW

**Comment by CatGarab** [comment]
[Top level requirements.txt for all the python packages (maybe one that just points at the other requirements.txt files?)](https://trello.com/c/b83ZVCVI/976-top-level-requirementstxt-for-all-the-python-packages-maybe-one-that-just-points-at-the-other-requirementstxt-files)

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+-r apis/app-api/python/requirements.txt
+-r apis/pumpkin-mcu-api/requirements.txt
+-r examples/pumpkin-mcu-api-example/requirements.txt
+-r examples/python-mission-application/requirements.txt
+-r examples/python-mission-framework/requirements.txt
+-r examples/python-service/requirements.txt
+-r hal/python-hal/i2c/requirements.txt
+-r libs/kubos-service/kubos_service/requirements.txt
```
Comment (plauche) [inline requirements.txt]: This requirements file is missing.

File: apis/app-api/python/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+mock==3.0.5
+requests==2.21.0
+responses==0.10.6
+setuptools==40.6.3
+toml==0.9.3
```
Comment (plauche) [inline apis/app-api/python/requirements.txt]: I had to move the `toml` version to `0.9.3.1`, it that `0.9.3` does not exist.

Comment (CatGarab) [reply apis/app-api/python/requirements.txt]: Interestingly, Buildroot is still pulling `0.9.3` successfully. So I guess we'll just have a slight version mismatch there

**Pull request merged**